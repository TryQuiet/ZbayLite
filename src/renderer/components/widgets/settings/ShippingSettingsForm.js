import React from 'react'
import PropTypes from 'prop-types'
import * as R from 'ramda'
import * as Yup from 'yup'
import { Formik, Form } from 'formik'
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import MenuItem from '@material-ui/core/MenuItem'
import Button from '@material-ui/core/Button'
import { withStyles } from '@material-ui/core/styles'

import TextField from '../../ui/TextField/TextField'
import SelectField from '../../ui/Select/SelectField'
import countryData from './countryData'
import UnfoldMore from '@material-ui/icons/UnfoldMore'

const styles = theme => ({
  fullWidth: {
    width: '100%'
  },
  item: {
    marginTop: 19
  },
  submitButton: {},
  label: {
    fontSize: 12,
    color: theme.palette.colors.black30
  },
  button: {
    marginTop: 14,
    height: 60,
    width: 126,
    fontSize: '0.9rem',
    backgroundColor: theme.palette.colors.zbayBlue
  },
  title: {
    marginBottom: 14
  },
  spacing: {
    marginLeft: 16
  },
  info: {
    marginBottom: 8
  }
})

export const formSchema = Yup.object().shape({
  firstName: Yup.string().required('Required'),
  lastName: Yup.string().required('Required'),
  street: Yup.string().required('Required'),
  country: Yup.string()
    .oneOf(R.keys(countryData))
    .required('Required'),
  region: Yup.string()
    .when('country', (country, schema) =>
      schema.oneOf(R.propOr([], country, countryData))
    )
    .required('Required'),
  city: Yup.string().required('Required'),
  postalCode: Yup.string().required('Required')
})

export const ShippingSettingsForm = ({
  classes,
  initialValues,
  handleSubmit
}) => {
  return (
    <Formik
      onSubmit={handleSubmit}
      validationSchema={formSchema}
      initialValues={initialValues}
    >
      {({ values, isSubmitting }) => (
        <Form className={classes.fullWidth}>
          <Grid container direction='column' alignItems='flex-start'>
            <Grid item className={classes.title}>
              <Typography variant='h3'>Shipping Address</Typography>
            </Grid>
            <Grid item className={classes.info}>
              <Typography variant='body2'>
                Your shipping address is saved on your computer, and sent in
                private to sellers only when you include it while making a
                purchase.{' '}
              </Typography>
            </Grid>

            <Grid item container direction='row' justify='space-between'>
              <Grid item xs>
                <Typography className={classes.label} variant='body2'>
                  First Name
                </Typography>
                <TextField
                  id='first-name'
                  name='firstName'
                  className={classes.field}
                  margin='none'
                  variant='outlined'
                  value={values.firstName}
                />
              </Grid>
              <Grid item xs className={classes.spacing}>
                <Typography className={classes.label} variant='body2'>
                  Last Name
                </Typography>
                <TextField
                  id='last-name'
                  name='lastName'
                  className={classes.field}
                  margin='none'
                  variant='outlined'
                  value={values.lastName}
                />
              </Grid>
            </Grid>
            <Grid
              item
              container
              direction='row'
              justify='space-between'
              className={classes.item}
            >
              <Grid item xs={12}>
                <Typography className={classes.label} variant='body2'>
                  Country
                </Typography>
                <SelectField
                  id='country'
                  name='country'
                  variant='outlined'
                  fullWidth
                  IconComponent={UnfoldMore}
                >
                  {R.keys(countryData).map(c => (
                    <MenuItem key={c} value={c}>
                      {c}
                    </MenuItem>
                  ))}
                </SelectField>
              </Grid>
            </Grid>
            <Grid
              item
              container
              direction='row'
              justify='space-between'
              className={classes.item}
            >
              <Grid item xs>
                <Typography className={classes.label} variant='body2'>
                  City
                </Typography>
                <TextField
                  id='city'
                  name='city'
                  className={classes.field}
                  margin='none'
                  variant='outlined'
                  value={values.city}
                />
              </Grid>
              <Grid item xs className={classes.spacing}>
                <Typography className={classes.label} variant='body2'>
                  Postal Code
                </Typography>
                <TextField
                  id='postal-code'
                  name='postalCode'
                  className={classes.field}
                  margin='none'
                  variant='outlined'
                  value={values.postalCode}
                />
              </Grid>
            </Grid>
            <Grid
              item
              container
              direction='row'
              justify='space-between'
              className={classes.item}
            >
              <Grid item xs={12}>
                <Typography className={classes.label} variant='body2'>
                  Region
                </Typography>
                <SelectField
                  id='region'
                  name='region'
                  variant='outlined'
                  fullWidth
                  IconComponent={UnfoldMore}
                >
                  {R.propOr([], values.country, countryData).map(r => (
                    <MenuItem key={r} value={r}>
                      {r}
                    </MenuItem>
                  ))}
                </SelectField>
              </Grid>
            </Grid>

            <Grid container item className={classes.item}>
              <Grid item xs={12}>
                <Typography className={classes.label} variant='body2'>
                  Street Address
                </Typography>
                <TextField
                  id='street-address'
                  name='street'
                  fullWidth
                  margin='none'
                  variant='outlined'
                  value={values.street}
                />
              </Grid>
            </Grid>
            <Grid container item>
              <Grid item xs={12} className={classes.submitButton}>
                <Button
                  variant='contained'
                  size='large'
                  color='primary'
                  type='submit'
                  fullWidth
                  disabled={isSubmitting}
                  className={classes.button}
                >
                  Save
                </Button>
              </Grid>
            </Grid>
          </Grid>
        </Form>
      )}
    </Formik>
  )
}

ShippingSettingsForm.propTypes = {
  classes: PropTypes.object.isRequired,
  initialValues: PropTypes.shape({
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
    street: PropTypes.string.isRequired,
    country: PropTypes.string.isRequired,
    region: PropTypes.string.isRequired,
    city: PropTypes.string.isRequired,
    postalCode: PropTypes.string.isRequired
  }).isRequired,
  handleSubmit: PropTypes.func.isRequired
}

ShippingSettingsForm.defaultProps = {
  initialValues: {
    firstName: '',
    lastName: '',
    street: '',
    country: '',
    region: '',
    city: '',
    postalCode: ''
  }
}

export default withStyles(styles)(ShippingSettingsForm)
