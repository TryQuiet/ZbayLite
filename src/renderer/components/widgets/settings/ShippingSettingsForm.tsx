import React from 'react'
import * as R from 'ramda'
import * as Yup from 'yup'
import { Formik, Form } from 'formik'
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import Button from '@material-ui/core/Button'
import { makeStyles } from '@material-ui/core/styles'

import TextField from '../../ui/TextField/TextField'
import countryData from './countryData'

const useStyles = makeStyles((theme) => ({
  fullWidth: {
    width: '100%'
  },
  item: {
    marginTop: 19
  },
  submitButton: {},
  label: {
    fontSize: 12,
    color: theme.palette.colors.black30
  },
  button: {
    marginTop: 14,
    height: 60,
    width: 126,
    fontSize: '0.9rem',
    backgroundColor: theme.palette.colors.zbayBlue
  },
  title: {
    marginBottom: 14
  },
  spacing: {
    marginLeft: 16
  },
  info: {
    marginBottom: 8
  }
}))

export const formSchema = Yup.object().shape({
  firstName: Yup.string().required('Required'),
  lastName: Yup.string().required('Required'),
  street: Yup.string().required('Required'),
  country: Yup.string()
    .oneOf(R.keys(countryData))
    .required('Required'),
  region: Yup.string()
    .when('country', (country, schema) =>
      schema.oneOf(R.propOr([], country, countryData))
    )
    .required('Required'),
  city: Yup.string().required('Required'),
  postalCode: Yup.string().required('Required')
})

interface ShippingSettingsFormProps {
  initialValues: {
    firstName: string
    lastName: string
    street: string
    country: string
    region: string
    city: string
    postalCode: string
  }
  handleSubmit: () => void
}

export const ShippingSettingsForm: React.FC<ShippingSettingsFormProps> = ({
  initialValues = {
    firstName: '',
    lastName: '',
    street: '',
    country: '',
    region: '',
    city: '',
    postalCode: ''
  },
  handleSubmit
}) => {
  const classes = useStyles({})
  return (
    <Formik
      onSubmit={handleSubmit}
      validationSchema={formSchema}
      initialValues={initialValues}
    >
      {({ values, isSubmitting }) => (
        <Form className={classes.fullWidth}>
          <Grid container direction='column' alignItems='flex-start'>
            <Grid item className={classes.title}>
              <Typography variant='h3'>Shipping Address</Typography>
            </Grid>
            <Grid item className={classes.info}>
              <Typography variant='body2'>
                Your shipping address is saved on your computer, and sent in
                private to sellers only when you include it while making a
                purchase.{' '}
              </Typography>
            </Grid>

            <Grid item container direction='row' justify='space-between'>
              <Grid item xs>
                <Typography className={classes.label} variant='body2'>
                  First Name
                </Typography>
                <TextField
                  id='first-name'
                  name='firstName'
                  margin='none'
                  variant='outlined'
                  value={values.firstName}
                />
              </Grid>
              <Grid item xs className={classes.spacing}>
                <Typography className={classes.label} variant='body2'>
                  Last Name
                </Typography>
                <TextField
                  id='last-name'
                  name='lastName'
                  margin='none'
                  variant='outlined'
                  value={values.lastName}
                />
              </Grid>
            </Grid>
            <Grid
              item
              container
              direction='row'
              justify='space-between'
              className={classes.item}
            >
              <Grid item xs={12}>
                <Typography className={classes.label} variant='body2'>
                  Country
                </Typography>
              </Grid>
            </Grid>
            <Grid
              item
              container
              direction='row'
              justify='space-between'
              className={classes.item}
            >
              <Grid item xs>
                <Typography className={classes.label} variant='body2'>
                  City
                </Typography>
                <TextField
                  id='city'
                  name='city'
                  margin='none'
                  variant='outlined'
                  value={values.city}
                />
              </Grid>
              <Grid item xs className={classes.spacing}>
                <Typography className={classes.label} variant='body2'>
                  Postal Code
                </Typography>
                <TextField
                  id='postal-code'
                  name='postalCode'
                  margin='none'
                  variant='outlined'
                  value={values.postalCode}
                />
              </Grid>
            </Grid>
            <Grid
              item
              container
              direction='row'
              justify='space-between'
              className={classes.item}
            >
              <Grid item xs={12}>
                <Typography className={classes.label} variant='body2'>
                  Region
                </Typography>
              </Grid>
            </Grid>

            <Grid container item className={classes.item}>
              <Grid item xs={12}>
                <Typography className={classes.label} variant='body2'>
                  Street Address
                </Typography>
                <TextField
                  id='street-address'
                  name='street'
                  fullWidth
                  margin='none'
                  variant='outlined'
                  value={values.street}
                />
              </Grid>
            </Grid>
            <Grid container item>
              <Grid item xs={12} className={classes.submitButton}>
                <Button
                  variant='contained'
                  size='large'
                  color='primary'
                  type='submit'
                  fullWidth
                  disabled={isSubmitting}
                  className={classes.button}
                >
                  Save
                </Button>
              </Grid>
            </Grid>
          </Grid>
        </Form>
      )}
    </Formik>
  )
}

export default ShippingSettingsForm
