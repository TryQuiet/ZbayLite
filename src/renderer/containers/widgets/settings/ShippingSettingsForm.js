import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import ShippingSettingsForm from '../../../components/widgets/settings/ShippingSettingsForm'
import identitySelectors from '../../../store/selectors/identity'
import identityHandlers from '../../../store/handlers/identity'

export const mapStateToProps = state => ({
  initialValues: identitySelectors.shippingData(state)
})

export const mapDispatchToProps = dispatch => bindActionCreators({
  handleSubmit: identityHandlers.epics.updateShippingData
}, dispatch)

export default connect(mapStateToProps, mapDispatchToProps)(ShippingSettingsForm)
