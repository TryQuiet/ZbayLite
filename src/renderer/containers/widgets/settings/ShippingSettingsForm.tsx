
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import ShippingSettingsForm from '../../../components/widgets/settings/ShippingSettingsForm'
import identityHandlers from '../../../store/handlers/identity'

export const mapStateToProps = _state => ({
  initialValues: ''
})

export const mapDispatchToProps = dispatch => bindActionCreators({
  handleSubmit: identityHandlers.epics.updateShippingData
}, dispatch)

export default connect(mapStateToProps, mapDispatchToProps)(ShippingSettingsForm)
