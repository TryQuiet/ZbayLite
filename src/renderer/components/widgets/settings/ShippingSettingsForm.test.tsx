/* eslint import/first: 0 */
import React from 'react'
import { shallow } from 'enzyme'

import { ShippingSettingsForm } from './ShippingSettingsForm'

describe('ShippingSettingsForm', () => {
  it('renders component', () => {
    const result = shallow(
      <ShippingSettingsForm
        initialValues={{
          firstName: 'Rumble',
          lastName: 'Fish',
          street: 'RumbleFish street',
          country: 'RumbleFish country',
          region: 'Fish Region',
          city: 'Fishville',
          postalCode: '1337-455'
        }}
        handleSubmit={jest.fn()}
      />
    )
    expect(result).toMatchSnapshot()
  })
})
