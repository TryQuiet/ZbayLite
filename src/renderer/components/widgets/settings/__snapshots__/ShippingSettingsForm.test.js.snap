// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShippingSettingsForm renders component 1`] = `
<Formik
  enableReinitialize={false}
  initialValues={
    Object {
      "city": "Fishville",
      "country": "RumbleFish country",
      "firstName": "Rumble",
      "lastName": "Fish",
      "postalCode": "1337-455",
      "region": "Fish Region",
      "street": "RumbleFish street",
    }
  }
  isInitialValid={false}
  onSubmit={[MockFunction]}
  validateOnBlur={true}
  validateOnChange={true}
  validationSchema={
    ObjectSchema {
      "_blacklist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "_conditions": Array [],
      "_defaultDefault": [Function],
      "_deps": Array [],
      "_excludedEdges": Array [],
      "_exclusive": Object {},
      "_mutate": undefined,
      "_nodes": Array [
        "postalCode",
        "city",
        "country",
        "region",
        "street",
        "lastName",
        "firstName",
      ],
      "_options": Object {
        "abortEarly": true,
        "recursive": true,
      },
      "_type": "object",
      "_typeError": [Function],
      "_whitelist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "fields": Object {
        "city": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "required": true,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
        "country": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "required": true,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {
              "Afghanistan",
              "Åland Islands",
              "Albania",
              "Algeria",
              "American Samoa",
              "Andorra",
              "Angola",
              "Anguilla",
              "Antarctica",
              "Antigua and Barbuda",
              "Argentina",
              "Armenia",
              "Aruba",
              "Australia",
              "Austria",
              "Azerbaijan",
              "Bahamas",
              "Bahrain",
              "Bangladesh",
              "Barbados",
              "Belarus",
              "Belgium",
              "Belize",
              "Benin",
              "Bermuda",
              "Bhutan",
              "Bolivia",
              "Bonaire, Sint Eustatius and Saba",
              "Bosnia and Herzegovina",
              "Botswana",
              "Bouvet Island",
              "Brazil",
              "British Indian Ocean Territory",
              "Brunei Darussalam",
              "Bulgaria",
              "Burkina Faso",
              "Burundi",
              "Cambodia",
              "Cameroon",
              "Canada",
              "Cape Verde",
              "Cayman Islands",
              "Central African Republic",
              "Chad",
              "Chile",
              "China",
              "Christmas Island",
              "Cocos (Keeling) Islands",
              "Colombia",
              "Comoros",
              "Congo, Republic of the (Brazzaville)",
              "Congo, the Democratic Republic of the (Kinshasa)",
              "Cook Islands",
              "Costa Rica",
              "Côte d'Ivoire, Republic of",
              "Croatia",
              "Cuba",
              "Curaçao",
              "Cyprus",
              "Czech Republic",
              "Denmark",
              "Djibouti",
              "Dominica",
              "Dominican Republic",
              "Ecuador",
              "Egypt",
              "El Salvador",
              "Equatorial Guinea",
              "Eritrea",
              "Estonia",
              "Ethiopia",
              "Falkland Islands (Islas Malvinas)",
              "Faroe Islands",
              "Fiji",
              "Finland",
              "France",
              "French Guiana",
              "French Polynesia",
              "French Southern and Antarctic Lands",
              "Gabon",
              "Gambia, The",
              "Georgia",
              "Germany",
              "Ghana",
              "Gibraltar",
              "Greece",
              "Greenland",
              "Grenada",
              "Guadeloupe",
              "Guam",
              "Guatemala",
              "Guernsey",
              "Guinea",
              "Guinea-Bissau",
              "Guyana",
              "Haiti",
              "Heard Island and McDonald Islands",
              "Holy See (Vatican City)",
              "Honduras",
              "Hong Kong",
              "Hungary",
              "Iceland",
              "India",
              "Indonesia",
              "Iran, Islamic Republic of",
              "Iraq",
              "Ireland",
              "Isle of Man",
              "Israel",
              "Italy",
              "Jamaica",
              "Japan",
              "Jersey",
              "Jordan",
              "Kazakhstan",
              "Kenya",
              "Kiribati",
              "Korea, Democratic People's Republic of",
              "Korea, Republic of",
              "Kosovo",
              "Kuwait",
              "Kyrgyzstan",
              "Laos",
              "Latvia",
              "Lebanon",
              "Lesotho",
              "Liberia",
              "Libya",
              "Liechtenstein",
              "Lithuania",
              "Luxembourg",
              "Macao",
              "Macedonia, Republic of",
              "Madagascar",
              "Malawi",
              "Malaysia",
              "Maldives",
              "Mali",
              "Malta",
              "Marshall Islands",
              "Martinique",
              "Mauritania",
              "Mauritius",
              "Mayotte",
              "Mexico",
              "Micronesia, Federated States of",
              "Moldova",
              "Monaco",
              "Mongolia",
              "Montenegro",
              "Montserrat",
              "Morocco",
              "Mozambique",
              "Myanmar",
              "Namibia",
              "Nauru",
              "Nepal",
              "Netherlands",
              "New Caledonia",
              "New Zealand",
              "Nicaragua",
              "Niger",
              "Nigeria",
              "Niue",
              "Norfolk Island",
              "Northern Mariana Islands",
              "Norway",
              "Oman",
              "Pakistan",
              "Palau",
              "Palestine, State of",
              "Panama",
              "Papua New Guinea",
              "Paraguay",
              "Peru",
              "Philippines",
              "Pitcairn",
              "Poland",
              "Portugal",
              "Puerto Rico",
              "Qatar",
              "Réunion",
              "Romania",
              "Russian Federation",
              "Rwanda",
              "Saint Barthélemy",
              "Saint Helena, Ascension and Tristan da Cunha",
              "Saint Kitts and Nevis",
              "Saint Lucia",
              "Saint Martin",
              "Saint Pierre and Miquelon",
              "Saint Vincent and the Grenadines",
              "Samoa",
              "San Marino",
              "Sao Tome and Principe",
              "Saudi Arabia",
              "Senegal",
              "Serbia",
              "Seychelles",
              "Sierra Leone",
              "Singapore",
              "Sint Maarten (Dutch part)",
              "Slovakia",
              "Slovenia",
              "Solomon Islands",
              "Somalia",
              "South Africa",
              "South Georgia and South Sandwich Islands",
              "South Sudan",
              "Spain",
              "Sri Lanka",
              "Sudan",
              "Suriname",
              "Swaziland",
              "Sweden",
              "Switzerland",
              "Syrian Arab Republic",
              "Taiwan",
              "Tajikistan",
              "Tanzania, United Republic of",
              "Thailand",
              "Timor-Leste",
              "Togo",
              "Tokelau",
              "Tonga",
              "Trinidad and Tobago",
              "Tunisia",
              "Turkey",
              "Turkmenistan",
              "Turks and Caicos Islands",
              "Tuvalu",
              "Uganda",
              "Ukraine",
              "United Arab Emirates",
              "United Kingdom",
              "United States",
              "United States Minor Outlying Islands",
              "Uruguay",
              "Uzbekistan",
              "Vanuatu",
              "Venezuela, Bolivarian Republic of",
              "Vietnam",
              "Virgin Islands, British",
              "Virgin Islands, U.S.",
              "Wallis and Futuna",
              "Western Sahara",
              "Yemen",
              "Zambia",
              "Zimbabwe",
            },
            "refs": Map {},
          },
          "_whitelistError": [Function],
          "tests": Array [
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
        "firstName": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "required": true,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
        "lastName": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "required": true,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
        "postalCode": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "required": true,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
        "region": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [
            Condition {
              "fn": [Function],
              "refs": Array [
                Reference {
                  "getter": [Function],
                  "isContext": false,
                  "isSibling": true,
                  "isValue": false,
                  "key": "country",
                  "map": undefined,
                  "path": "country",
                },
              ],
            },
          ],
          "_deps": Array [
            "country",
          ],
          "_exclusive": Object {
            "required": true,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
        "street": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "required": true,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
      },
      "tests": Array [],
      "transforms": Array [
        [Function],
      ],
    }
  }
>
  <Component />
</Formik>
`;
